src/api.cpp

```diff
@@ -163,6 +163,16 @@ struct api_query_string : public std::string {
     size_t size_with_null() const { return size() + 1; }
 };
 
+bool out_of_order_device_support(const cl_device_id devices, cl_int err) {
```

Comment:
What's the idea with `cl_int err` here? If the goal is to be able to have the caller being able to have the value, it needs to be a reference.